### DISCUSSION TITLE
defined field cannot get the value

### DISCUSSION THREAD

**Discussion by zhujintao** [kubernetes-sigs/kubebuilder#3091] [Support] [answered]
How to get spec.appConf.core key name  b2b-plus 、 bps-core

###
type:
type DeploysSpec struct {
	AppConf map[string]map[string][]string `json:"appConf,omitempty"`
}
###
yaml:
  appConf:
    core:
      b2b-plus:
      bps-core:
###
spec:
  appConf:
    core: {}

**Comment by camilamacedo86** [comment, marked as answer]
Hi @zhujintao,

Sorry, but the info provided here does not seems enough for we try to help you out.
What do you mean with  `How to get spec.appConf.core` ? Would you like to access the spec in the reconciliation/controller?
If so, see the code generated for the deploy image plugin: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3-with-deploy-image/controllers/busybox_controller.go#L239

Also, I'd like to encourage you check out the plugin doc: 
And then, try to create an example with to check the code for better understand how you should/can develop the solutions. https://book.kubebuilder.io/plugins/deploy-image-plugin-v1-alpha.html

On top of that, I'd like to recommend you read the [Tutorial: Building CronJob](https://book.kubebuilder.io/cronjob-tutorial/cronjob-tutorial.html#tutorial-building-cronjob) at least to have a better understand.

**Reply by zhujintao** [reply]
tks, please take a look at this problem
https://github.com/kubernetes-sigs/kubebuilder/issues/3093

**Comment by camilamacedo86** [comment]
Closing since it is answered.
